### DISCUSSION TITLE
Streaming Response in Chunks

### DISCUSSION THREAD

**Discussion by chrisbll971** [hyperium/hyper#3460] [Q&A]
Hello,

I was wondering if it is possible to send a response in chunks back to the caller in a stream?

For example:
1. Send back a part of the file (for example, some metadata about the file)
2. Wait a few seconds
3. Send back the rest of the file

Thanks

**Comment by seanmonstar** [comment]
Yes, you can. The bodies in hyper support streaming. You can wrap an existing `impl Stream`, or use a channel, or some other mechanism.
